test/cases/regression-old-worker-new-composer.sh

```diff
@@ -54,123 +59,362 @@ priority=${priority}
-    grep -qE "Job [0-9a-fA-F-]+ finished"
+greenprint "Creating dummy rpm and repository to test payload_repositories"
+sudo dnf install -y rpm-build createrepo
+DUMMYRPMDIR=$(mktemp -d)
+DUMMYSPECFILE="$DUMMYRPMDIR/dummy.spec"
+PAYLOAD_REPO_PORT="9999"
+PAYLOAD_REPO_URL="http://localhost:9999"
+pushd "$DUMMYRPMDIR"
 
-# Did the compose finish with success?
-if [[ $COMPOSE_STATUS != FINISHED ]]; then
-    echo "Something went wrong with the compose. 😢"
-    exit 1
+cat <<EOF > "$DUMMYSPECFILE"
+#----------- spec file starts ---------------
+Name:                   dummy
+Version:                1.0.0
+Release:                0
+BuildArch:              noarch
+Vendor:                 dummy
+Summary:                Provides %{name}
+License:                BSD
+Provides:               dummy
+%description
+%{summary}
+%files
+EOF
+
+mkdir -p "DUMMYRPMDIR/rpmbuild"
+rpmbuild --quiet --define "_topdir $DUMMYRPMDIR/rpmbuild" -bb "$DUMMYSPECFILE"
+
+mkdir -p "$DUMMYRPMDIR/repo"
+cp "$DUMMYRPMDIR"/rpmbuild/RPMS/noarch/*rpm "$DUMMYRPMDIR/repo"
+pushd "$DUMMYRPMDIR/repo"
+createrepo .
+sudo python3 -m http.server "$PAYLOAD_REPO_PORT" &
+KILL_PIDS+=("$!")
+popd
+popd
+
+greenprint "Installing aws client tools"
+if ! hash aws; then
+  echo "Using 'awscli' from a container"
+  sudo ${CONTAINER_RUNTIME} pull ${CONTAINER_IMAGE_CLOUD_TOOLS}
+
+  AWS_CMD="sudo ${CONTAINER_RUNTIME} run --rm \
+    -e AWS_ACCESS_KEY_ID=${V2_AWS_ACCESS_KEY_ID} \
+    -e AWS_SECRET_ACCESS_KEY=${V2_AWS_SECRET_ACCESS_KEY} \
+    -v ${WORKDIR}:${WORKDIR}:Z \
+    ${CONTAINER_IMAGE_CLOUD_TOOLS} aws --region $AWS_REGION --output json --color on"
+else
+  echo "Using pre-installed 'aws' from the system"
+  AWS_CMD="aws --region $AWS_REGION --output json --color on"
 fi
+$AWS_CMD --version
+
+greenprint "Preparing request"
+REQUEST_FILE="${WORKDIR}/request.json"
+ARCH=$(uname -m)
+CLOUD_PROVIDER="aws.s3"
+IMAGE_TYPE="guest-image"
 
-tee "dnf-json-request.json" <<EOF
+# This removes dot from VERSION_ID.
+# ID == rhel   && VERSION_ID == 8.6 => DISTRO == rhel-86
+# ID == centos && VERSION_ID == 8   => DISTRO == centos-8
+# ID == fedora && VERSION_ID == 35  => DISTRO == fedora-35
+DISTRO="$ID-${VERSION_ID//./}"
+
+cat > "$REQUEST_FILE" << EOF
 {
-    "command": "dump",
-    "arguments": {
-        "repos": [
-            {
-                "name": "fedora",
-                "id": "blep-2",
-                "metalink": "https://mirrors.fedoraproject.org/metalink?repo=fedora-35&arch=x86_64",
-                "check_gpg": true
-            }
-        ],
-        "arch": "x86_64",
-        "module_platform_id": "platform:f35"
+  "distribution": "$DISTRO",
+  "customizations": {
+    "payload_repositories": [
+      {
+        "baseurl": "$PAYLOAD_REPO_URL"
+      }
+    ],
+    "packages": [
+      "postgresql",
+      "dummy"
+    ],
+    "users": [
+      {
+        "name": "user1",
+        "groups": ["wheel"]
+      },
+      {
+        "name": "user2"
+      }
+    ]
+  },
```

Comment:
So user customization is something we technically don't do in the service (yet). But it can't hurt to have it here.

Reply:
ah yeah good point.  Copied from the old test.  Like you said, can't hurt to cover it, unless there's any chance it interferes with the real test.